src/index.js

```diff
@@ -21,26 +21,63 @@ const packageDefinition = protoLoader.loadSync(
     }
 );
 
-const grpcProtoDescriptor = grpc.loadPackageDefinition(packageDefinition);
-
 // This is the JavaScript's best way of cloning an object.
 const jsonPackageDefinition = JSON.parse(JSON.stringify(packageDefinition));
-const jsonService = jsonPackageDefinition["clarifai.api.V2"];
-for (const methodName in jsonService) {
-    if (jsonService.hasOwnProperty(methodName)) {
-        jsonService[methodName].requestSerialize = (argument) => argument;
-        jsonService[methodName].responseDeserialize = (data) => data;
+
+/*
+ * When sending the data over the JSON channel, if a concept doesn't have a value, the server defaults the value
+ * to 1. With the gRPC channel however, all missing fields get a default value of 0. To do the same in JSON channel
+ * what is done in the default gRPC channel, explicit 0 is set.
+ *
+ * Ideally this would be done dynamically by inspecting the "cl_default_float" custom option of the proto field, but
+ * I'm not sure how to find out whether a field has this set or not.
+ */
```

Comment:
Is this because we have that custom jsonpb package on backend side?

Reply:
This also seems like a user error and we should mimic any old defaults on the server side. If they want a positive concept they should set to True and all examples should have that no?

Reply:
@zeiler Currently, when using the gRPC channel and not setting `value` inside of a `concept` object, the server-side will assume you mean 0 and mark the concept as negative (in all three languages). This code also makes sure the JSON channel behaves the same.

In the code examples, everywhere we want to set a positive concept, we explicitly set the value to 1.

Reply:
yeah I think it's that `cl_default_float` option on the proto that does this in our json parser. Over time I want to remove all those types of field options though as it's weird to have defaults that are different than the proto defaults (which are if nothing is provided it defaults to the zero value). That's why all the examples have value = 1.0 when it's needed which is good. So the gRPC channel is doing the correct long term thing.

Reply:
i think i see what you mean now with this code, ok this sounds good as if it doesn't currently have the field it sets value = 0 on json, that's great as it makes that cl_default_float skipped on that backend.